draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -130,7 +138,86 @@ This document uses "service" and "workload" interchangeably. Otherwise, all term
 
 # Application Level Service To Service Authentication {#app-level}
 
-## The WIMSE ID Token
+## The Workload Identity Token {#to-wit}
+
+The Workload Identity Token (WIT) is a JWS {{RFC7515}} signed JWT {{RFC7519}} that represents the identity of a workload.
+It is issued by the Identity Server and binds a public key to the workload identity.
+A WIT MUST contain the following:
+
+* in the JOSE header:
+    * `alg`: An identifier for a JWS asymmetric digital signature algorithm
+     (registered algorithm identifiers are listed in the IANA JOSE Algorithms registry {{IANA.JOSE.ALGS}}). The value `none` MUST NOT be used.
+    * `typ`: the WIT is explicitly typed, as recommended in {{Section 3.11 of RFC8725}}, using the `wimse-id+jwt` media type.
+* in the JWT claims:
+    * `iss`: The issuer of the token, which is the Identity Server, represented by a URI.
+    * `sub`: The subject of the token, which is the identity of the workload, represented by a URI.
+    * `exp`: The expiration time of the token (as defined in {{Section 4.1.4 of RFC7519}}).
+    * `jti`: A unique identifier for the token.
+    * `cnf`: A confirmation claim containing the public key of the workload using the `jwk` member as defined in {{Section 3.2 of RFC7800}}.
+
+An example WIT might look like this (all examples, of course, are non-normative and with line breaks and extra space for readability):
+
+~~~ jwt
+eyJ0eXAiOiJ3aW1zZS1pZCtqd3QiLCJhbGciOiJFUzI1NiIsImtpZCI6Ikp1bmUgNSJ9.
+eyJpc3MiOiJ3aW1zZTovL2V4YW1wbGUuY29tL3RydXN0ZWQtY2VudHJhbC1hdXRob3Jpd
+HkiLCJleHAiOjE3MTc2MTI0NzAsInN1YiI6IndpbXNlOi8vZXhhbXBsZS5jb20vc3BlY2
+lmaWMtd29ya2xvYWQiLCJqdGkiOiJ4LV8xQ1RMMmNjYTNDU0U0Y3diX18iLCJjbmYiOns
+iandrIjp7Imt0eSI6Ik9LUCIsImNydiI6IkVkMjU1MTkiLCJ4IjoiX2FtUkMzWXJZYkho
+SDFSdFlyTDhjU21URE1oWXRPVVRHNzhjR1RSNWV6ayJ9fX0.rOSUMR8I5WhM5C704l3iV
+dY0zFqxhugJ8Jo2xo39G7FqUTbwTzAGdpz2lHp6eL1M486XmRgl3uyjj6R_iuzNOA
+~~~
+
+The decoded JOSE header of the WIT from the example above is shown here:
+
+~~~ json
+{
+ "typ": "wimse-id+jwt",
+ "alg": "ES256",
+ "kid": "June 5"
+}
+~~~
+
+The decoded JWT claims of the WIT from the example above are shown here:
+
+~~~ json
+{
+ "iss": "wimse://example.com/trusted-central-authority",
+ "exp": 1717612470,
+ "sub": "wimse://example.com/specific-workload",
+ "jti": "x-_1CTL2cca3CSE4cwb__",
+ "cnf": {
+  "jwk": {
```

Comment:
This key would shortly be used to sign its own JWTs (or HTTP messages). Doesn't it make sense to include a "kid" for it, too?

Reply:
Thought about that but (without having gone deep into the details) didn't see the need for a kid here as the relationship will be implied and the key will be there in the WIT.

Reply:
As we discussed, `jwk` allows you to include a `kid`. So I'm fine with not mentioning it.